CommunicationLayer/CommDeviceControl/UdpConnectionService.cpp

```diff
@@ -14,38 +26,83 @@ UdpConnectionService::~UdpConnectionService()
 
 void  UdpConnectionService::setMulticastNetwork(QHostAddress groupAddress, quint16 port)
 {
-    groupAddress_ = groupAddress;
-    port_ = port;
+    //groupAddress_ = groupAddress;
+    //port_ = port;
+
+
+
 }
 
-bool UdpConnectionService::connectToDataSource()
+void UdpConnectionService::setupChannel()
 {
-    if (groupAddress_.isNull() || port_ == 0)
-    {
-        emit connectionFailed("Group address and port number is not set!");
-        return false;
-    }
+    quint32 i = 0;
 
-    if (socket_.bind(QHostAddress::AnyIPv4, port_, QUdpSocket::ShareAddress) &&
-            socket_.joinMulticastGroup(groupAddress_))
-    {
-        emit connectionSucceeded();
-        return true;
-    }
-    else
+    do
     {
-        emit connectionFailed(socket_.errorString());
-        return false;
+        if (i++)
+        {
+            qWarning() << "UdpMessageForwarder: Attempting to reconnect" << ipAddress_ << udpPort_;
+        }
+
+        try
+        {
+            channel_ = AmqpClient::Channel::Create(ipAddress_.toStdString(), udpPort_);
+        }
+        catch (std::exception&)
```

Comment:
As a general rule, catch all statements are bad. This is because an exception should only be catch if it's clear what the course of action should be for that exception,. So it's not that this catch statement is bad, but a specific exception should be caught here.